server/homebrew.api.js

```diff
@@ -6,13 +6,59 @@ const GoogleActions = require('./googleActions.js');
 const Markdown = require('../shared/naturalcrit/markdown.js');
 const yaml = require('js-yaml');
 const asyncHandler = require('express-async-handler');
+const { nanoid } = require('nanoid');
 
 // const getTopBrews = (cb) => {
 // 	HomebrewModel.find().sort({ views: -1 }).limit(5).exec(function(err, brews) {
 // 		cb(brews);
 // 	});
 // };
 
+const findBrew = async (brewId, accessType, googleObjectId)=>{
+	let id = brewId;
+	let googleId = googleObjectId;
+	if(id.length > 12) {
+		googleId = id.slice(0, -12);
+		id = id.slice(-12);
+	}
+	let brew = await HomebrewModel.get(accessType === 'edit' ? { editId: id } : { shareId: id })
+		.catch((err)=>{
+			if(googleId) {
+				console.warn(`Unable to find document stub for ${accessType}Id ${id}`);
+			} else {
+				console.warn(err);
+			}
+		});
+
+	let googleBrew;
+	if(googleId || brew?.googleId) {
+		googleBrew = await GoogleActions.getGoogleBrew(googleId || brew?.googleId, id, accessType)
+			.catch((err)=>{
+				console.warn(err);
+			});
+		if(!brew && !googleBrew) throw 'Brew not found in database or Google Drive';
+		brew = brew ? _.merge(brew, googleBrew) : googleBrew;
```

Comment:
In this case `googleBrew` does have a possibility of not existing. If the call to `GoogleActions.getGoogleBrew` throws an exception, it will be caught by the `.catch` method above. Since that's a different method scope, I have it writing the error to the log and ending the scope, which is the same as doing a return there.

The check there can have googleBrew as undefined, and since we've already looked for the brew in the db and also couldn't find it, the brew is unknown to us so we throw the exception.

I've attached a console example:

![image](https://user-images.githubusercontent.com/4909307/163089247-fa3a4461-9aef-4d33-809d-e348079a7f85.png)

Reply:
Mostly there's just a lot of checking for existence of brew/googleBrew and I was trying to cut down on any redundant checks for legibility. If this is the easiest way to avoid an error then let's keep it. I'll take another look and see if there's any other cleanups we can make.

Reply:
Yeah that's entirely fair and if you do find anything please let me know! 

I think I may have added a few checks exactly like the above now in the deletion, update, and new brew methods. I also adjusted the return types of some of the google action methods to account for that. Some that were previously void now return the object id or just `true`.